ui/components/FormDialog.vue

```diff
@@ -0,0 +1,155 @@
+<template>
+  <simple-dialog
+    v-model="value"
+    class="form-dialog"
+    :class="`form-dialog_${mode}`"
+    :actions="actions"
+    :back-disabled="isFirstTab"
+    :details-route="detailsRoute"
+    :height="height"
+    :is-valid="isTabValid"
+    :on-submit="submit"
+    :title="computedTitle"
+    :width="width"
+    @back="goToPreviousTab"
+    @next="goToNextTab"
+  >
+    <template #sidebar>
+      <vertical-tabs
+        v-model:active-tab-key="activeTabKey"
+        :tabs="tabs"
+        :linear="mode === 'wizard'"
+      />
+    </template>
+    <template #default>
+      <slot :name="activeTabKey" />
+    </template>
+  </simple-dialog>
+</template>
+
+<script setup>
+import { ref, computed, watch } from 'vue';
+
+import SimpleDialog from '~/components/SimpleDialog.vue';
+import VerticalTabs from '~/components/VerticalTabs.vue';
+
+const props = defineProps({
+  title: {
+    type: String,
+    required: true,
+  },
+  tabs: {
+    type: Array,
+    required: true,
+  },
+  mode: {
+    type: String,
+    required: true,
+    validator(value) {
+      return ['wizard', 'edit'].includes(value);
+    },
+  },
+  form: {
+    type: Object,
+    required: true,
+  },
+  height: {
+    type: String,
+    default: '600px',
+  },
+  width: {
+    type: String,
+    default: '800px',
+  },
+  onSubmit: {
+    type: Function,
+    default: () => {},
+  },
+  detailsRoute: {
+    type: Object,
+    default: null,
+  },
+  rules: {
+    type: Object,
+    default: () => {},
+  },
+  submitLabel: {
+    type: String,
+    default: '',
+  },
+});
+
+const value = defineModel({
+  type: Boolean,
+  default: false,
+});
+
+const activeTabKey = ref('');
+const activeTabIdx = computed(() => props.tabs.findIndex((tab) => tab.key === activeTabKey.value));
+const activeTab = computed(() => props.tabs.find((tab) => tab.key === activeTabKey.value));
+const isFirstTab = computed(() => activeTabIdx.value === 0);
+const isLastTab = computed(() => activeTabIdx.value === props.tabs.length - 1);
+const isSummaryTab = computed(() => activeTabKey.value === 'summary');
+const isSubmittableTab = computed(() => activeTab.value.submittable);
+
+const computedTitle = computed(() => {
+  if (props.mode === 'edit') return props.title;
+  if (isSummaryTab.value) return `${props.title} – Summary`;
+  return `${props.title} – Step ${activeTabIdx.value + 1}`;
+});
+
+const isTabValid = computed(() => {
+  if (activeTab.value.includes) {
```

Comment:
why the if is on the existence of that function? is there a prettier way to do it?

Reply:
Hm, the only thing I can think of is to have a computed to check if the active tab has `includes`.
Something like `const activeTabHasRules = computed(() => Boolean(activeTab.value.includes));`, and then use it here. Wdyt?

Reply:
Check the updated version: https://github.com/cloudblue/connect-bi-reporter/pull/31/commits/58e2dc8fb1a3c031bf3ca6b5791f8206d04b4f14#diff-611371e20b38fc91f0d07b7e105f5dc7ab8b9d65174a21700f15c3c664a35419R104